### PR TITLE
Adds new tags to PhpdocAlignFixer

### LINKED ISSUE #5290: PhpdocAlignFixer breaks with psalm-param or phpstan-param
For example, when running against:

```
    /**
     * @param         string       $source        source package name
     * @param         string       $sourceVersion source package version (pretty version ideally)
     * @param         string       $description   link description (e.g. requires, replaces, ..)
     * @phpstan-param Link::TYPE_* $description
     * @param         array        $links         array of package name => constraint mappings
     * @return Link[]
     */
    public function parseLinks($source, $sourceVersion, $description, $links)
    {
    }
```

The output is:

```
    /**
     * @param string $source        source package name
     * @param string $sourceVersion source package version (pretty version ideally)
     * @param string $description   link description (e.g. requires, replaces, ..)
     * @phpstan-param Link::TYPE_* $description
     * @param  array  $links array of package name => constraint mappings
     * @return Link[]
     */
    public function parseLinks($source, $sourceVersion, $description, $links)
```

So we see that things are aligned above and below the phpstan-param, but it kinda broke alignment. This makes sense as per the [configuration of phpdoc_align](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.16/doc/rules/phpdoc/phpdoc_align.rst#tags) which only lists other tags, but I do wonder why the list of tags is restricted. Why can't I add any tag name in there and have it align? Or alternatively, could php-cs-fixer support these phpstan/psalm-param tags and others out of the box?

### PR DESCRIPTION
**Pull request by niklam**
New PHPDoc tags

  - @param-out
  - @template
  - @template-covariant
  - @extends
  - @implements
  - @deprecated
  - @internal
  - @readonly
  - @no-named-arguments
  - @psalm-allow-private-mutation
  - @psalm-assert
  - @psalm-assert-if-false
  - @psalm-if-this-is
  - @psalm-assert-if-true
  - @psalm-consistent-constructor
  - @psalm-consistent-templates
  - @psalm-external-mutation-free
  - @psalm-ignore-falsable-return
  - @psalm-ignore-nullable-return
  - @psalm-ignore-var
  - @psalm-immutable
  - @psalm-import-type
  - @psalm-internal
  - @psalm-method
  - @psalm-mutation-free
  - @psalm-param
  - @psalm-param-out
  - @psalm-property
  - @psalm-property-read
  - @psalm-property-write
  - @psalm-pure
  - @psalm-readonly
  - @psalm-readonly-allow-private-mutation
  - @psalm-require-extends
  - @psalm-require-implements
  - @psalm-return
  - @psalm-seal-properties
  - @psalm-suppress SomeIssueName
  - @psalm-taint-*
  - @psalm-trace
  - @psalm-type
  - @psalm-var
  - @phpstan-var
  - @phpstan-param
  - @phpstan-return
  - @phpstan-template
  - @phpstan-template-covariant
  - @phpstan-extends
  - @phpstan-implements

Closes #5290

### CODE REVIEW

**Comment by niklam** [comment]
As an additional feature I'd suggest to add new parameter for `phpdoc_align`: `align_all_tags` which would accept `bool`. If set to `true`, this would override `tags`, and align all pre-defined tags.

I already did this for my own use for this PR, but I think it should be a separate PR as it's obviously a separate feature from this PR.

**Comment by niklam** [comment]
Oh, and some of the tags should be defined as simple tags, if this approach is something you guys would be happy with.

**Comment by niklam** [comment]
I don't understand why the tests fail. Why would those tags just disappear? 😮

File: src/Fixer/Phpdoc/PhpdocAlignFixer.php [hunk 1]
```diff
@@ -92,13 +160,19 @@ public function configure(array $configuration): void
...
         $indent = '(?P<indent>(?:\ {2}|\t)*)';
 
+        // e.g. @deprecated
+        if ([] !== $simpleTags) {
+            $types[] = '(?P<tag>'.implode('|', $simpleTags).')';
+        }
+
         // e.g. @param <hint> <$var>
         if ([] !== $tagsWithNameToAlign) {
             $types[] = '(?P<tag>'.implode('|', $tagsWithNameToAlign).')\s+(?P<hint>(?:'.TypeExpression::REGEX_TYPES.')?)\s+(?P<var>(?:&|\.{3})?\$\S+)';
```
Comment (WinterSilence) [inline src/Fixer/Phpdoc/PhpdocAlignFixer.php:178]: > I don't understand why the tests fail. 

similar sub-pattern `<tag>`

Comment (WinterSilence) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php:178]: + `(?P<var>(&|\.{3})\s*\$[\w]+\S+)`

Comment (WinterSilence) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php:178]: if `<hint>` not required you must use `(?P<hint>('.TypeExpression::REGEX_TYPES.')\s+)*?`

Comment (niklam) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php:178]: I'm not sure I understand what you mean.

Comment (WinterSilence) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php:178]: @niklam you add sub-pattern `tag` in line 173, but sub-pattern with same name already defined in line 178, pattern name or combine them

Comment (niklam) [reply src/Fixer/Phpdoc/PhpdocAlignFixer.php:178]: Oh, right. Thanks. I renamed the newly added sub-pattern to `simpleTag`, and added processing for it to 432.

**Comment by VincentLanglet** [comment]
There is a lot of tags (at least for phpstan) which should be supported and missing  in this PR.
For instance phpstan-method, or phpstan-property.

I think the best would be to stop restricting the value passed to the config.
Others PHPDoc Fixer are not restricting them.

I implemented this in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6564

**Pull request closed without merging**